lib/ansible/modules/cloud/amazon/guardduty_ip_list.py

```diff
@@ -0,0 +1,235 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: guardduty_ip_list
+short_description: Manage Trusted/Threat IP list for AWS GuardDuty.
+description:
+    - Manage a Trusted/Threat IP list assigned to a AWS GuardDuty detector.
+author: "Aaron Smith (@slapula)"
+version_added: "2.7"
+requirements: [ 'botocore', 'boto3' ]
+options:
+  name:
+    description:
+    - A user-friendly name that is displayed in all finding generated by activity that involves
+      IP addresses included in this list.
+    required: true
+  list_type:
+    description:
+    - Whether the list should be exist or not on the detector.
+    required: true
+    choices: ['trusted', 'threat']
+  state:
+    description:
+    - Whether the list should be exist or not on the detector.
+    default: 'present'
+    choices: ['present', 'absent']
+  enabled:
+    description:
+    - Whether the list should be enabled or disabled.
+    type: bool
+    default: true
+  detector_id:
+    description:
+    - The unique ID of the detector that you want to update.
+    required: true
+  format:
+    description:
+    - The format of the file that contains the IP list.
+    choices: ['TXT', 'STIX', 'OTX_CSV', 'ALIEN_VAULT', 'PROOF_POINT', 'FIRE_EYE']
+  location:
+    description:
+    - The URI of the file that contains the IP list.
+extends_documentation_fragment:
+    - ec2
+    - aws
+'''
+
+
+EXAMPLES = r'''
+- name: create GuardDuty detector
+  guardduty_threatlist:
+    state: present
+    enabled: true
+
+- name: disable GuardDuty detector
+  guardduty_threatlist:
+    state: present
+    enabled: false
+
+- name: delete GuardDuty detector
+  guardduty_threatlist:
+    state: absent
+'''
+
+
+RETURN = r'''#'''
+
+import os
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, AWSRetry
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+
+def ip_list_exists(client, module, result):
+    try:
+        if module.params.get('list_type') == 'trusted':
+            sets = client.list_ip_sets(
+                DetectorId=module.params.get('detector_id')
+            )
+            for i in sets['IpSetIds']:
+                ti_set = client.get_ip_set(
+                    DetectorId=module.params.get('detector_id'),
+                    IpSetId=i
+                )
+                if ti_set['Name'] == module.params.get('name'):
+                    result['trusted_set_id'] = i
```

Comment:
I feel like passing around a reference to a dict complicates things unnecessarily and makes the code less obvious. That isn't a big deal when the module is so small, but if more options are introduced and the module grows it can become more of a pain. We recently removed this pattern from ec2_group and I'd prefer not to add it in new places. It looks like the meaningful information this function should provide is either to return the set ID or None.